dbt_worker/app.py

```diff
@@ -16,5 +18,42 @@
 
 app.config_from_object(celeryconfig)
 
+LIVENESS_FILE = Path("/tmp/worker_heartbeat")
+
+
+class LivenessProbe(bootsteps.StartStopStep):
+    requires = {"celery.worker.components:Timer"}
+
+    def __init__(self, _, **kwargs):
+        self.requests = []
+        self.tref = None
+
+    def start(self, worker):
+        self.tref = worker.timer.call_repeatedly(
+            1.0,
+            self.update_heartbeat_file,
+            (worker,),
+            priority=10,
+        )
+
+    def stop(self, _):
+        LIVENESS_FILE.unlink(missing_ok=True)
+
+    def update_heartbeat_file(self, _):
+        LIVENESS_FILE.touch()
+
+
+@worker_ready.connect
+def worker_ready(**_):
+    LIVENESS_FILE.touch()
```

Comment:
cool usage of Path.touch!

Reply:
Each Celery worker runs this right? Could there be any threading problems with multiple workers interacting with the same file?